src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,141 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final Object[] EMPTY_ELEMENTDATA = {};
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ZERO = 0;
+    private int size;
+    private T[] values;
 
+    public ArrayList() {
+        this.size = ZERO;
+        this.values = (T[]) new Object[DEFAULT_CAPACITY];
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T value) {
+        if (size >= values.length) {
+            values = grow(values.length);
+        }
+        values[size] = backEmptyForNull(value);
+        size++;
+    }
 
+    @Override
+    public void add(T value, int index) throws ArrayListIndexOutOfBoundsException {
+        checkBoundsIndex((index == size && size != 0) ? (index - 1) : index);
+        int newSize = size + 1;
+        if (newSize >= values.length) {
+            values = grow(values.length);
+        }
+        T[] tempArray = (T[]) new Object[values.length];
+        for (int i = 0; i < index; i++) {
+            tempArray[i] = values[i];
+        }
+        tempArray[index] = backEmptyForNull(value);
+        for (int i = index + 1; i < newSize; i++) {
+            tempArray[i] = values[i - 1];
+        }
```

Comment:
Use approach according to checklist:
https://github.com/mate-academy/jv-array-list/blob/master/checklist.md#use-systemarraycopy-to-move-your-array-elements